### PR TITLE
Shrink installed size and separate startup scripts

### PR DESCRIPTION
**Pull request by smacfarlane**
copy_hab_stuff brought in way more than was necessary and allowed the exporter to work when it should have been broken. To resolve this, KERNEL and SYSTEM_PKGS variables were added. These are things needed to boot the system, but aren't necessarily provided by the user packages, and lay the groundwork for custom kernels. An export of core/redis is ~200 MB now, instead of nearly 2G. 

This also separates hab startup from the startup script.  This was initially to allow the udhcpc to run at boot, since the `hab sup bash` stopped execution of init. uhdcpc is unable to obtain an address when run from inittab and enters an endless loop, so is currently disabled on boot.

### CODE REVIEW

File: bin/hab-pkg-image.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
-
+KERNEL="core/linux"
+INIT=(core/iproute2 core/busybox-static core/util-linux core/coreutils)
```
Comment (elliott-davis) [inline bin/hab-pkg-image.sh]: I think since you have `core/coreutils` in the plan.sh you don't need it here

File: plan.sh [hunk 2]
```diff
@@ -4,7 +4,6 @@ pkg_version="0.1.0"
 pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
 pkg_license=('Apache-2.0')
 pkg_deps=(
-  core/linux
   core/bash
```
Comment (elliott-davis) [inline plan.sh]: I think we can remove this dep since everything is linked against busybox

File: bin/hab-pkg-image.sh (hunk 1 above)
Comment (elliott-davis) [inline bin/hab-pkg-image.sh]: oh nevermind - I think I see what you're doing - ignore me on this one

Comment (smacfarlane) [inline bin/hab-pkg-image.sh]: Yeah,  I think long term we'll want to create a meta-package, similar to hab-backline, that brings in all the system packages.

**Pull request merged**